src/mad_spark_alt/core/robust_json_handler.py

```
@@ -0,0 +1,197 @@
+            if _validate_json(data, expected_keys):
+                return data
+        except json.JSONDecodeError:
+            pass
+    
+    # Strategy 5: Extract key-value pairs manually
+    if expected_keys:
+        extracted = _extract_key_values(response, expected_keys)
+        if extracted:
+            return extracted
+    
+    logger.warning(f"Failed to extract JSON from response: {response[:200]}...")
```

Ruff (0.11.9)

Logging statement uses f-string

(G004)